src/enchcracker/cracker/JavaSingleSeedCracker.java

```diff
@@ -51,7 +51,7 @@ public void resetCracker() {
 	@Override
 	public void firstInput(int bookshelves, int slot1, int slot2, int slot3) {
 		abortRequested.set(false);
-		final int threadCount = Runtime.getRuntime().availableProcessors() - 1; // always leave one for OS
+		final int threadCount = Runtime.getRuntime().availableProcessors() - 1 != 0 ? Runtime.getRuntime().availableProcessors() - 1 : 1; // always leave one for OS; ternary op to ensure that it is never zero
```

Comment:
```suggestion
		final int threadCount = Math.max(Runtime.getRuntime().availableProcessors() - 1, 1); // always leave one for OS but ensure at least 1 processor is used
```

To me this is easier to read than the ternary and doesn't duplicate the calculation.